### PR TITLE
#8441: Add note about squash merging

### PR DESCRIPTION
**Pull request by tt-rkim**

### CODE REVIEW

**Comment by tt-rkim** [comment]
Hi everyone, we're slowly rolling out squash and merge via the GitHub UI. I tagged a subset of our engineers to review this new feature.

I don't believe this should cause merge commits to go in, as it should use `--ff-only` internally.

Let us know if we can update the documentation in anyway.

I know @bbradelTT had some good notes for newcomer devs to be familiar with our GitHub process. We will update the notes accordingly

**Comment by bbradelTT** [comment]
When you squash and merge, does it ask you for a new commit message?

**Comment by tt-rkim** [comment]
> When you squash and merge, does it ask you for a new commit message?

Yes, and I'm thinking we tell people to supply one that follows our format and summarizes your issue.

**Comment by bbradelTT** [comment]
> > When you squash and merge, does it ask you for a new commit message?
> 
> Yes, and I'm thinking we tell people to supply one that follows our format and summarizes your issue.

Then be more explicit. Something like:
If you use squashing, when you enter a new commit message, ensure that your commit message summarizes your change and follows our required format. ...

**Review by bbradelTT — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -436,12 +436,21 @@ Sometimes signatures will point directly to the issue (eg, dispatch), though som
...
   history.
 
+- You can use either of the following methods to merge your branch on the
+  GitHub UI:
+  - Squash and merge
+  - Rebase and merge
+
+  If you use squashing, when GitHub asks you to enter a new commit message,
+  ensure that your commit message follows our required format as outlined above
+  in this section. Failure to do so is a violation of our standards.
```
Comment (ayerofieiev-tt) [inline CONTRIBUTING.md:446]: @tt-rkim , I never saw github asking. My expectation is that it uses PR Title for a new commit and PR description for an extended commit description.

**Review by ayerofieiev-tt — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (tt-rkim) [reply CONTRIBUTING.md:446]: Screenshot 2024-05-14 at 12 48 38 PM
These fields are editable once you click Squash and Merge
Do you not see that?

Comment (ayerofieiev-tt) [reply CONTRIBUTING.md:446]: Oh, thats right. Thank you!

**Review by abhullar-tt — APPROVED** [review]

**Review by tt-aho — APPROVED** [review]

**Pull request merged**